test/e2e/clusterclass_changes_test.go

```diff
@@ -46,3 +46,29 @@ var _ = Describe("When testing ClusterClass changes [ClusterClass]", func() {
 		})
 	})
 })
+
+var _ = Describe("When testing ClusterClass changes [vcsim] [supervisor] [PR-Blocking] [ClusterClass]", func() {
```

Comment:
q: is it intentional to run this test on vcsim? (the corresponding govmomi test runs on real infrastructure)
q: is it intentional to run it as a PR blocking? (the corresponding govmomi test is not blocking)

Reply:
And what's our criteria to label a test case as PR blocking? In this case, I'll fix it and align with the govmomi test as not blocking

Reply:
> How to decide whether a case should be labeled to run with a) or b)

It is our decision if to run a test using vcsim or real infrastructure, with the caveat that the option to run tests on vcsim was recently introduced.
Generally speaking I would prefer to run tests on real infrastructure, and use vcsim only if we have constraints about time or resources (which is not the current case I think).

So I will use a) for this case.

> And what's our criteria to label a test case as PR blocking? 

We usually consider PR blocking the minimal set of tests that we run on every PR.
Considering that this test is more focused on validating if a regression happens in CAPI than validating CAPV changes, I'm +1 to not have this as a PR blocking test.

Also, please note that as a maintainer we can dynamically expand the set or E2E to pass on each PR with the /test prow command, so it is ok to have very few E2E tests as PR blocking .

Reply:
Agree, we don't have to run this test with vcsim and we shouldn't make it PR blocking.

This then leaves: `When testing ClusterClass changes [supervisor] [ClusterClass]`

But if we add an additional test like this, this test will run twice for govmomi.

This is because the govmomi test runs basically without any focus/skip: https://prow.k8s.io/prowjob?prowjob=0ed84718-7065-4355-9c4a-2242f7616e04
And supervisor runs with focus on [supervisor]: https://prow.k8s.io/prowjob?prowjob=207d878f-5aa4-47a2-b29b-1e2a81808393

So what we have to do instead is to add [supervisor] to the existing test

But now you'll need a bit of branching for ModifyMachineDeploymentInfrastructureMachineTemplateFields, this can be easily done with `testMode == GovmomiTestMode`

Reply:
Thanks for the clarification! Now it's more clear to me.
And this is even better to leverage the existing test case instead of adding duplicate code :-)